File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left = 0
        visited = {}
        ret = 0
        for right, c in enumerate(s):
            if c in visited:
                left = max(left, visited[c] + 1)
            visited[c] = right
            ret = max(ret, right-left+1)
        return ret

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int curr_min = prices[0];
        int ret = 0;
        for (int i=1; i<prices.size(); i++) {
            if (prices[i]<=curr_min) {
                curr_min = prices[i];
            } else {
                ret = max(ret, prices[i]-curr_min);
            }
        }
        return ret;
    }
};
